presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp

```diff
@@ -1924,6 +1927,120 @@ core::PlanFragment VeloxQueryPlanConverterBase::toVeloxQueryPlan(
   return planFragment;
 }
 
+namespace {
+
+bool rowHasTimestampWithTz(const RowTypePtr row) {
+  for (auto child : row->children()) {
+    if (child->kindEquals(TIMESTAMP_WITH_TIME_ZONE())) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+bool exprHasTimestampWithTz(const std::vector<core::TypedExprPtr>& exprs) {
+  for (auto expr : exprs) {
+    if (expr->type()->kindEquals(TIMESTAMP_WITH_TIME_ZONE())) {
+      return true;
+    }
+    if (exprHasTimestampWithTz(expr->inputs())) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+bool planHasTimestampWithTimeZone(const core::PlanNodePtr planNode) {
+  // Do output types have timestamp with tz ?
+  if (rowHasTimestampWithTz(planNode->outputType())) {
+    return true;
+  }
+
+  // Is it a projectNode with timestamp with tz ?
+  if (auto projectNode =
+          std::dynamic_pointer_cast<const core::ProjectNode>(planNode)) {
+    if (exprHasTimestampWithTz(projectNode->projections())) {
+      return true;
+    }
+  }
+
+  // Is it a filterNode with timestamp with tz ?
+  if (auto filterNode =
+          std::dynamic_pointer_cast<const core::FilterNode>(planNode)) {
+    if (exprHasTimestampWithTz({filterNode->filter()})) {
+      return true;
+    }
+  }
+
+  // Is it one of the joinNodes with timestamp with tz ?
+  if (auto joinNode =
+          std::dynamic_pointer_cast<const core::HashJoinNode>(planNode)) {
+    if (exprHasTimestampWithTz({joinNode->filter()})) {
+      return true;
+    }
+  }
+
+  if (auto joinNode =
+          std::dynamic_pointer_cast<const core::AbstractJoinNode>(planNode)) {
+    if (exprHasTimestampWithTz({joinNode->filter()})) {
+      return true;
+    }
+  }
+
+  if (auto joinNode =
+          std::dynamic_pointer_cast<const core::MergeJoinNode>(planNode)) {
+    if (exprHasTimestampWithTz({joinNode->filter()})) {
+      return true;
+    }
+  }
+
+  if (auto joinNode =
+          std::dynamic_pointer_cast<const core::NestedLoopJoinNode>(planNode)) {
+    if (exprHasTimestampWithTz({joinNode->joinCondition()})) {
+      return true;
+    }
+  }
+
+  // Is it tablescan with remaining filter ?
+  if (auto tableScanNode =
+          std::dynamic_pointer_cast<const core::TableScanNode>(planNode)) {
+    if (auto hiveTableHandle =
+            std::dynamic_pointer_cast<connector::hive::HiveTableHandle>(
+                tableScanNode->tableHandle())) {
+      if (exprHasTimestampWithTz({hiveTableHandle->remainingFilter()})) {
+        return true;
+      }
+    }
+  }
```

Comment:
AggregationNode::aggregates_::call is a CallTypedExprPtr that can contain a LambdaTypedExprPtr as input. Should we check it too?